encoding.c

```diff
@@ -810,11 +824,22 @@ rb_enc_autoload(rb_encoding *enc)
     return i;
 }
 
+bool
+rb_enc_autoload_p(rb_encoding *enc)
+{
+    int idx = ENC_TO_ENCINDEX(enc);
+    RUBY_ASSERT(rb_enc_from_index(idx) == enc);
```

Comment:
FYI: @nobu

Reply:
`enc->loaded` doesn't exist. `loaded` is stored in `struct rb_encoding_entry` not in `rb_encoding`.

Putting it in `rb_encoding` would be much simpler, but I assumed patching `onigmo` wasn't acceptable.